### DISCUSSION TITLE
为什么早期的 Windows 需要整理碎片 - 面向信仰编程 · /whys-the-design-windows-defragmentation

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#644] [Announcements]
https://draveness.me/whys-the-design-windows-defragmentation/

**Comment by yuchanns** [comment]
Windows现在还是需要的，只是Win10有定时任务会自动整理，稍微提高了用户体验。

不过我也有四年没怎么正经用Windows了~ **I Love Mac!**

文末问题，为什么Mac不需要？其实Mac以前也需要(HFS)，只不过早就自动整理了。到了High Sierra，苹果的文件系统改成了APFS (Apple File System) ，就不需要了。

稍微查了下，一方面，依然有自动整理功能；另一方面，APFS在写入时采用写入连续数据块，以及队列延迟写入，减少了碎片化的情况。

Linux采用的ext4目前查到的说法是因为不同文件之间在写入的时候刻意分散之间留下了比较大的空间，所以存储的时候需要分散存储文件碎片的情况比较少，在需要的时候也会自动整理碎片。

具体细节就看左神后续文章了:stuck_out_tongue_winking_eye:

另外，前几天无意中发现了这个博客的[主题](https://github.com/jrutheiser/hugo-lithium-theme)，哈哈！然后我也稍微改了改拿来自己[用](https://yuchanns.xyz/)了

**Comment by OhYee** [comment]
> @wuqinqiang
> 老哥你画图都是用啥工具的

帮作者回一下

https://draveness.me/sketch-and-sketch/

**Comment by luohy15** [comment]
请问作者考虑过给博客文章加上索引（例如 [Go 语言设计与实现](https://draveness.me/golang/) 页面右侧的索引），或者把博客原文放在 GitHub 上吗？
对于一些较长的文章，例如 [调度系统设计精要](https://draveness.me/system-design-scheduler/) [内存管理设计精要](https://draveness.me/system-design-memory-management/) ，虽然给出了思维导图，但是个人认为如果有可跳转的索引，阅读体验会更好一些。

**Comment by baichuan0898** [comment]
“软盘都只能整盘的写入” 的说法是不对的。97年的时候我就开始用电脑了，那个时候的电脑上软驱是标配。软盘用起来跟现在的U盘是差不多，可以存放很多独立文件，也可以只对单个文件进行读写。不需要每次读写文件都要整盘覆盖的。除了软盘容量比现在U盘小很多（3.5 寸软盘容量只有 1.44 MB），以及读写速度上软盘要慢很多以外，其他方面用起来感觉跟现代U盘没差。

**Comment by draveness** [comment]
> @baichuan0898
> “软盘都只能整盘的写入” 的说法是不对的。97年的时候我就开始用电脑了，那个时候的电脑上软驱是标配。软盘用起来跟现在的U盘是差不多，可以存放很多独立文件，也可以只对单个文件进行读写。不需要每次读写文件都要整盘覆盖的。除了软盘容量比现在U盘小很多（3.5 寸软盘容量只有 1.44 MB），以及读写速度上软盘要慢很多以外，其他方面用起来感觉跟现代U盘没差。

这里确实有问题，我改一下

**Comment by draveness** [comment]
> @luohy15
> 请问作者考虑过给博客文章加上索引（例如 [Go 语言设计与实现](https://draveness.me/golang/) 页面右侧的索引），或者把博客原文放在 GitHub 上吗？
> 对于一些较长的文章，例如 [调度系统设计精要](https://draveness.me/system-design-scheduler/) [内存管理设计精要](https://draveness.me/system-design-memory-management/) ，虽然给出了思维导图，但是个人认为如果有可跳转的索引，阅读体验会更好一些。
> 

加起来比较麻烦啊，之前加过后来懒了

**Comment by lostpg** [comment]
> @luohy15
> 请问作者考虑过给博客文章加上索引（例如 [Go 语言设计与实现](https://draveness.me/golang/) 页面右侧的索引），或者把博客原文放在 GitHub 上吗？
> 对于一些较长的文章，例如 [调度系统设计精要](https://draveness.me/system-design-scheduler/) [内存管理设计精要](https://draveness.me/system-design-memory-management/) ，虽然给出了思维导图，但是个人认为如果有可跳转的索引，阅读体验会更好一些。
> 

chromium 内核的浏览器的话，试试这个 [给网页加浮动 TOC 的插件](https://chrome.google.com/webstore/detail/smart-toc/lifgeihcfpkmmlfjbailfpfhbahhibba)。

**Comment by draveness** [comment]
> @lostpg
> 
> chromium 内核的浏览器的话，试试这个 [给网页加浮动 TOC 的插件](https://chrome.google.com/webstore/detail/smart-toc/lifgeihcfpkmmlfjbailfpfhbahhibba)。

这个非常不错啊，不过要是 Hugo 有类似的插件就更好了
